parser.js

```diff
@@ -1,273 +1,51 @@
-                msg[1] = obj.start;
-                msg[2] = obj.start;
-            }
-            if (obj.type === "UNBOUND_ATTRIBUTE_VALUE"){
-                msg[0] = Mustache.render(errorJSON.UNBOUND_ATTRIBUTE_VALUE, { 'error': obj.value });
-                msg[1] = obj.interval.start;
-                msg[2] = obj.interval.end;
-            }
-            if (obj.type === "UNTERMINATED_OPEN_TAG"){
-                msg[0] = Mustache.render(errorJSON.UNTERMINATED_OPEN_TAG, { 'error': obj.openTag.name });
-                msg[1] = obj.openTag.start;
-                msg[2] = obj.openTag.end;
-            }
-            for(var i = msg[1]; i <= msg[2]; i++){
-                output += input[i];
-            }
-            msg[5] = output;
+    function render(html, context) {
+        var replaceText = {
+            closeTag: context.closeTag && context.closeTag.name ? context.closeTag.name : "",
+            openTag: context.openTag && context.openTag.name ? context.openTag.name : "",
+            attribute: context.attribute && context.attribute.name && context.attribute.name.value ? context.attribute.name.value : "",
+            cssValue: context.cssValue && context.cssValue.value ? context.cssValue.value : "",
+            cssSelector: context.cssSelector && context.cssSelector.selector ?  context.cssSelector.selector : "",
+            cssProperty: context.cssProperty && context.cssProperty.property ? context.cssProperty.property : "",
+            cssKeyword: context.cssKeyword && context.cssKeyword.value ? context.cssKeyword.value : "",
+            name: context.name || "",
+            error: context.error && context.error.msg ? context.error.msg : "",
+            value: context.value || ""
+        };
 
-            //Finds the line number for the start of the error
-            for(i = 0; i <= (msg[1] + 1); i++)
-            {
-                if(input[i] === "\n")
-                {
-                    lineCount += 1;
-                    lineBeginStart = i;
-                }
-                parsedText += input[i];
-                charCount++;
-            }
-            //line number for start of error
-            msg[3] = lineCount;
+        return html
+                .replace(/\[\[closeTag\.name\]\]/, replaceText.closeTag)
+                .replace(/\[\[openTag\.name\]\]/, replaceText.openTag)
+                .replace(/\[\[attribute\.name\.value\]\]/, replaceText.attribute)
+                .replace(/\[\[cssValue\.value\]\]/, replaceText.cssValue)
+                .replace(/\[\[cssSelector\.selector\]\]/, replaceText.cssSelector)
+                .replace(/\[\[cssProperty\.property\]\]/, replaceText.cssProperty)
+                .replace(/\[\[cssKeyword\.value\]\]/, replaceText.cssKeyword)
+                .replace(/\[\[name\]\]/, replaceText.name)
+                .replace(/\[\[error\.msg\]\]/, replaceText.error)
+                .replace(/\[\[value\]\]/, replaceText.value);
```

Comment:
Could you not use Mustache or some other template rendering to do this, vs. having to regex all this?  Isn't that the point of why they did it in that style?